### PR TITLE
Se133 ssr historical data process

### PR DESCRIPTION
**Pull request by ChunzhaoLi**
Hi @RGriffin-Quest , could you please help to review the help changes? Thanks.

### CODE REVIEW

File: pages/SpotlightEnterprise/enterprise_backend_federation_add.md [hunk 1]
```diff
@@ -51,5 +51,39 @@ Spotlight Clients in the federation have access to all configuration templates i
...
+    exec spotlight_merge_data_in_same_ssr 2  
+2. Execute the spotlight_merge_data_in_same_ssr with the @To_domain_id value is 3 to merge the historical data from other domains to the current domain 3.
+    exec spotlight_merge_data_in_same_ssr 3
+	
+After these two steps, the historical data will been merged to the Current used domains 2 and 3.
+
+
+### Historical data and the current data are in different database
+
+In SSMS, expand the SSR database, there is a stored procedure - spotlight_merge_data_in_diff_ssr, it has a parameter: @HistoricalSSRDatabaseName. This procedure can merge the data from an historical SSR database to the currently used SSR database. Below is an example to show how to use this procedure.
```
Comment (RGriffin-Quest) [inline pages/SpotlightEnterprise/enterprise_backend_federation_add.md:75]: Change to:
Using SSMS expand the SSR database and locate the stored spotlight_merge_data_in_diff_ssr. This procedure should be used when you want to move data from an old domain to a currently used domain in a different SSR database. In the SSR the domain is equivalent to the Diagnostic Server host name. This procedure has a parameter called @HistoricalSSRDatabaseName which is the name of the new SSR database that you want to transfer the data to. Below is an example of how to use this procedure.

File: pages/SpotlightEnterprise/enterprise_backend_federation_add.md [hunk 2]
```diff
@@ -51,5 +51,39 @@ Spotlight Clients in the federation have access to all configuration templates i
...
+
+For example, we have two SSR databases - SSR_Old and SSR_New, some of connections appear in both SSR_Old and SSR_New with the same connection name and some of the connections only exist in SSR_Old. Now we want to transfer the data from the SSR_Old to SSR_New:
```
Comment (RGriffin-Quest) [inline pages/SpotlightEnterprise/enterprise_backend_federation_add.md:77]: Change to:
For example say that you have two SSR databases - SSR_Old and SSR_New. Some of connections appear in both SSR_Old and SSR_New with the same connection name and some of the connections only exist in SSR_Old. Now you want to transfer the data from the SSR_Old to SSR_New.

File: pages/SpotlightEnterprise/enterprise_backend_federation_add.md [hunk 3]
```diff
@@ -51,5 +51,39 @@ Spotlight Clients in the federation have access to all configuration templates i
...
+1. If the connection name in both SSR database, merge to the exist connection in SSR_New;
+2. If the connection name not in SSR_New, add the connection and the data into SSR_New;
```
Comment (RGriffin-Quest) [inline pages/SpotlightEnterprise/enterprise_backend_federation_add.md:79]: I don't understand what these 2 sentences mean. If this telling the user that they need to do two steps or that the script will do two these things.

File: pages/SpotlightEnterprise/enterprise_backend_federation_add.md [hunk 4]
```diff
@@ -51,5 +51,39 @@ Spotlight Clients in the federation have access to all configuration templates i
...
+
+### Historical data and the current data are in different database
+
+In SSMS, expand the SSR database, there is a stored procedure - spotlight_merge_data_in_diff_ssr, it has a parameter: @HistoricalSSRDatabaseName. This procedure can merge the data from an historical SSR database to the currently used SSR database. Below is an example to show how to use this procedure.
+
+For example, we have two SSR databases - SSR_Old and SSR_New, some of connections appear in both SSR_Old and SSR_New with the same connection name and some of the connections only exist in SSR_Old. Now we want to transfer the data from the SSR_Old to SSR_New:
+1. If the connection name in both SSR database, merge to the exist connection in SSR_New;
+2. If the connection name not in SSR_New, add the connection and the data into SSR_New;    
+
+If SSR_Old and SSR_New are in the same database instance, execute spotlight_merge_data_in_diff_ssr with the @HistoricalSSRDatabaseName value is [SSR_Old].
```
Comment (RGriffin-Quest) [inline pages/SpotlightEnterprise/enterprise_backend_federation_add.md:81]: Change to:
If the SSR_Old and SSR_New databases are in the same instance then execute spotlight_merge_data_in_diff_ssr with the @HistoricalSSRDatabaseName parameter set to [SSR_Old]

File: pages/SpotlightEnterprise/enterprise_backend_federation_add.md [hunk 5]
```diff
@@ -51,5 +51,39 @@ Spotlight Clients in the federation have access to all configuration templates i
...
+exec spotlight_merge_data_in_diff_ssr "[SSR_Old]"
+
+If SSR_Old is in another database instance with an IP 10.154.10.10, spotlight_merge_data_in_diff_ssr with the @HistoricalSSRDatabaseName value is [10.154.10.10].[SSR_Old].
```
Comment (RGriffin-Quest) [inline pages/SpotlightEnterprise/enterprise_backend_federation_add.md:84]: Change to:
If the SSR_Old database is in another instance with an IP address of 10.154.10.10 then execute  spotlight_merge_data_in_diff_ssr with the @HistoricalSSRDatabaseName parameter set to [10.154.10.10].[SSR_Old].

File: pages/SpotlightEnterprise/enterprise_backend_federation_add.md [hunk 6]
```diff
@@ -51,5 +51,39 @@ Spotlight Clients in the federation have access to all configuration templates i
...
+exec spotlight_merge_data_in_diff_ssr "[10.154.10.10].[SSR_Old]"
+
+Note:  SSR databases used to merge should has the same COLLATE settings.
```
Comment (RGriffin-Quest) [inline pages/SpotlightEnterprise/enterprise_backend_federation_add.md:87]: Change to:
Note:  Both SSR databases involved in the merge must has the same collation settings.

**Comment by ChunzhaoLi** [comment]
Hi @RGriffin-Quest , Thank you very much for your patience in helping me correct the document. I have changed the document according to the comments. Could you please help to review it again? Thanks a lot!

File: pages/SpotlightEnterprise/enterprise_backend_federation_add.md [hunk 7]
```diff
@@ -51,39 +51,42 @@ Spotlight Clients in the federation have access to all configuration templates i
...
-## SSR historical data process while making a federation
+## Reorganizing SSR historical data after creating a federation
+
+If you have created a Federation that monitors connections that were previously monitored by one Diagnostic Server and that Diagnostic Server was storing data to the Spotlight Statistics Repository (SSR) and you want to preserve the historical data in the SSR take the following steps. This should only be done after the Federation has been writing to the SSR for one day to ensure that all connections have been re-created in the new SSR.
 
 ### Check the Diagnostic Server list and the data statistics information for each connection
-In SSMS, select the SSR database, there is a stored procedure - spotlight_merge_data_check_info, execute it, will get these information.
 
+We should check how many Diagnostic Servers in the SSR database and the time period and amount of data collected per connection.
+Using the SSMS select the SSR database locate the stored procedure spotlight_merge_data_check_info and execute it to see this information.
```
Comment (RGriffin-Quest) [inline pages/SpotlightEnterprise/enterprise_backend_federation_add.md:61]: Change to:
Using SSMS select the SSR database, locate the stored procedure spotlight_merge_data_check_info and execute it to see this information.

File: pages/SpotlightEnterprise/enterprise_backend_federation_add.md [hunk 8]
```diff
@@ -51,39 +51,42 @@ Spotlight Clients in the federation have access to all configuration templates i
...
 
-## SSR historical data process while making a federation
+## Reorganizing SSR historical data after creating a federation
+
+If you have created a Federation that monitors connections that were previously monitored by one Diagnostic Server and that Diagnostic Server was storing data to the Spotlight Statistics Repository (SSR) and you want to preserve the historical data in the SSR take the following steps. This should only be done after the Federation has been writing to the SSR for one day to ensure that all connections have been re-created in the new SSR.
 
 ### Check the Diagnostic Server list and the data statistics information for each connection
-In SSMS, select the SSR database, there is a stored procedure - spotlight_merge_data_check_info, execute it, will get these information.
 
+We should check how many Diagnostic Servers in the SSR database and the time period and amount of data collected per connection.
```
Comment (RGriffin-Quest) [inline pages/SpotlightEnterprise/enterprise_backend_federation_add.md:60]: Change to:
Use the following steps to check how many Diagnostic Servers are in the SSR database and the time period and amount of data collected per connection.

**Comment by ChunzhaoLi** [comment]
Hi @RGriffin-Quest , thanks a lot.  I have corrected the document according to your comments. Could you please help to review it again? Thanks.

**Pull request merged**